pkg/eventing/v1alpha1/client.go

```diff
@@ -118,3 +131,70 @@ func (c *knEventingClient) Namespace() string {
 func updateTriggerGvk(obj runtime.Object) error {
 	return util.UpdateGroupVersionKindWithScheme(obj, v1alpha1.SchemeGroupVersion, scheme.Scheme)
 }
+
+// TriggerBuilder is for building the trigger
+type TriggerBuilder struct {
+	trigger *v1alpha1.Trigger
+}
+
+// NewTriggerBuilder for building trigger object
+func NewTriggerBuilder(name string) *TriggerBuilder {
+	return &TriggerBuilder{trigger: &v1alpha1.Trigger{
+		ObjectMeta: meta_v1.ObjectMeta{
+			Name: name,
+		},
+	}}
+}
+
+// NewTriggerBuilderFromExisting for building the object from existing Trigger object
+func NewTriggerBuilderFromExisting(tr *v1alpha1.Trigger) *TriggerBuilder {
+	return &TriggerBuilder{trigger: tr.DeepCopy()}
+}
+
+// Broker to set the broker of trigger object
+func (b *TriggerBuilder) Broker(broker string) *TriggerBuilder {
+	if broker != "" {
+		b.trigger.Spec.Broker = broker
+	}
+	return b
+}
+
+// Filters to set the filters of trigger object
+func (b *TriggerBuilder) Filters(filters map[string]string) *TriggerBuilder {
+	if filters != nil {
+		triggerFilterAttributes := v1alpha1.TriggerFilterAttributes(filters)
+		b.trigger.Spec.Filter = &v1alpha1.TriggerFilter{
+			Attributes: &triggerFilterAttributes,
+		}
+	}
+	return b
+}
+
+// UpdateFilters to update the filters of trigger object
+func (b *TriggerBuilder) UpdateFilters(toUpdate map[string]string, toRemove []string) *TriggerBuilder {
```

Comment:
I wouldn't do any removal or change in the builder. Its about creating an immutable object, so would prefer to have just a `Filter()` method (as in my PR ;-) if you don't mind I would update that later.

Reply:
Find. I don't mind. Please update it in a separate PR.